### PR TITLE
Sketch base class and TE and TS implementations

### PR DESCRIPTION
**Pull request by RagnarGrootKoerkamp**

### CODE REVIEW

File: python/base.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    d.sort(key=lambda tup: tup[0])
+    return d
+
+
+sketchparams_spec = [('A', nb.int32), ('t', nb.int32), ('D', nb.int32), ('normalize', nb.bool_)]
+
+
+@jitclass(sketchparams_spec)
+class SketchParams:
+    def __init__(self, A, t, D, normalize=True):
```
Comment (danieldanciu) [inline python/base.py:45]: Can you document the A, t, D params? Only people familar with our notation would know what they mean.

File: python/tensor_sketch.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            # Normalization factor.
+            n = seq.len()
+            nct = nb.float64(1)
+            for i in range(self.t):
+                nct = nct * (n - i) / (i + 1)
+            full_sketch[-1] /= nct
+        sketch = np.array([x for x in full_sketch[-1]], dtype=nb.float32)
+        return SketchedSequence(seq, sketch)
+
+    # Returns the sketch for the given t as frequencies.
```
Comment (danieldanciu) [inline python/tensor_sketch.py:50]: which t?

**Review by danieldanciu — APPROVED** [review]

File: python/base.py (hunk 1 above)
Comment (RagnarGrootKoerkamp) [reply python/base.py:45]: the next PR will do this.

File: python/tensor_sketch.py (hunk 2 above)
Comment (RagnarGrootKoerkamp) [reply python/tensor_sketch.py:50]: hmm obsolete comment it seems.

**Comment by danieldanciu** [comment]
Ok - same comment appears in the other file.

**Pull request merged**